cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -96,7 +97,7 @@ func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesA
 	// Construct a descriptor for the current state of the world
 	beforeState := ClusterState{
 		KubeVersion:     clusterVersionStr,
-		DNSVersion:      dns.GetKubeDNSVersion(clusterVersion),
+		DNSVersion:      dns.GetDNSVersion(clusterVersion, kubeadmconstants.KubeDNS),
```

Comment:
Can you add a TODO here to say that we should make CoreDNS available here as well.
Not needed for the initial PR though